crates/client/src/lib.rs

```diff
@@ -539,17 +540,29 @@ impl<R: RegistryStorage, C: ContentStorage> Client<R, C> {
         }
 
         if let Some(from) = self.registry.load_checkpoint().await? {
-            if from.as_ref().checkpoint.log_length < ts_checkpoint.as_ref().checkpoint.log_length {
-                self.api
-                    .prove_log_consistency(
-                        ConsistencyRequest {
-                            from: from.as_ref().checkpoint.log_length,
-                            to: ts_checkpoint.as_ref().checkpoint.log_length,
-                        },
-                        Cow::Borrowed(&from.as_ref().checkpoint.log_root),
-                        Cow::Borrowed(&ts_checkpoint.as_ref().checkpoint.log_root),
-                    )
-                    .await?;
+            let from_log_length = from.as_ref().checkpoint.log_length;
+            let to_log_length = ts_checkpoint.as_ref().checkpoint.log_length;
+
+            match from_log_length.cmp(&to_log_length) {
+                Ordering::Greater => {
+                    return Err(ClientError::CheckpointLogLengthRewind {
+                        from: from_log_length,
+                        to: to_log_length,
+                    })
+                }
+                Ordering::Less => {
+                    self.api
+                        .prove_log_consistency(
+                            ConsistencyRequest {
+                                from: from_log_length,
+                                to: to_log_length,
+                            },
+                            Cow::Borrowed(&from.as_ref().checkpoint.log_root),
+                            Cow::Borrowed(&ts_checkpoint.as_ref().checkpoint.log_root),
+                        )
+                        .await?
+                }
+                Ordering::Equal => {}
```

Comment:
Does this need to check for total checkpoint equality? (I'm really not sure; just want to be sure that the registry can't swap in another checkpoint with the same length)

Reply:
Possible it would be caught during other parts of the inclusion proofs. But better to make it an explicit check and error. Pushed an update.